File: backend/app/schemas/metodo_pago.py
METODOS_PAGO = {
    "efectivo", "tarjeta_credito", "tarjeta_debito",
    "transferencia", "cheque", "vale_vista",
    "credito_simple", "otros",
}

PLAZO_FORZADO_CERO = {"efectivo", "tarjeta_debito", "tarjeta_credito"}
PLAZO_OBLIGATORIO = {"credito_simple"}


def validate_metodo_plazo(metodo: str | None, plazo: int) -> None:
    if metodo is None:
        return
    if metodo not in METODOS_PAGO:
        raise ValueError(f"metodo_pago inválido. Opciones: {sorted(METODOS_PAGO)}")
    if metodo in PLAZO_FORZADO_CERO and plazo != 0:
        raise ValueError(f"'{metodo}' requiere plazo_dias = 0")
    if metodo in PLAZO_OBLIGATORIO and plazo == 0:
        raise ValueError(f"'{metodo}' requiere plazo_dias > 0")

File: frontend/src/lib/metodo_pago.ts
export const METODOS_PAGO = [
  'efectivo', 'tarjeta_credito', 'tarjeta_debito',
  'transferencia', 'cheque', 'vale_vista',
  'credito_simple', 'otros',
] as const

export type MetodoPago = typeof METODOS_PAGO[number]

export const METODO_PAGO_LABELS: Record<string, string> = {
  efectivo: 'Efectivo',
  tarjeta_credito: 'T. Crédito',
  tarjeta_debito: 'T. Débito',
  transferencia: 'Transferencia',
  cheque: 'Cheque',
  vale_vista: 'Vale Vista',
  credito_simple: 'Crédito',
  otros: 'Otros',
}

export const PLAZO_FORZADO_CERO = new Set(['efectivo', 'tarjeta_debito', 'tarjeta_credito'])
export const PLAZO_OBLIGATORIO = new Set(['credito_simple'])

export const PLAZO_OPTIONS = [
  { value: 0, label: 'Al contado' },
  { value: 30, label: '30 días' },
  { value: 60, label: '60 días' },
  { value: 90, label: '90 días' },
]

export function formatMetodoPlazo(metodo: string | null | undefined, plazo: number): string {
  if (!metodo) return ''
  const label = METODO_PAGO_LABELS[metodo] ?? metodo
  if (plazo === 0) return `${label} al contado`
  return `${label} a ${plazo} días`
}

export function isPlazoForzadoCero(metodo: string): boolean {
  return PLAZO_FORZADO_CERO.has(metodo)
}
